test/cqlpy/test_tablets.py

```diff
@@ -181,7 +181,38 @@ def test_describe_initial_tablets(cql, skip_without_tablets):
             "AND TABLETS = { 'initial' : 1 }"
     with new_test_keyspace(cql, ksdef) as keyspace:
         desc = cql.execute(f"DESCRIBE KEYSPACE {keyspace}")
-        assert "and tablets = {'initial': 1}" in desc.one().create_statement.lower()
+        assert "and tablets = {'enabled': true, 'initial': 1}" in desc.one().create_statement.lower()
+
+
+# Test to ensure metadata is printed when tablets are enabled during schema description, regardless of the use of WITH INTERNALS.
+#
+# Reproduces https://github.com/scylladb/scylladb/issues/22866
+@pytest.mark.parametrize("with_internals", [True, False])
+@pytest.mark.parametrize("tablets", [None, True, False])
+@pytest.mark.parametrize("initial", [None, 0, 7])
+def test_describe_tablets_internals(cql, with_internals, tablets, initial):
+    tablets_str = (f"'enabled': {tablets}" if tablets is not None else "").lower()
+    initial_str = f"'initial': {initial}" if initial is not None else ""
```

Comment:
It is quite pointless to check with enabled=false and 3 different "initial" options. In fact, I expect it to be an **error** to specify initial if enabled=false...

So instead of the 9 remaining combinations of with_internals and tablets I suggest you just paramterize the pairs that are really allowed. I expect only 4 combinations: Tablets disabled, and tablets enabled with 3 settings for initial.

Finally, "initial=None" should be the same as "initial=0". I'm not sure if you intended to verify this in this test, if not, you don't need to test it here.

Reply:
> Finally, "initial=None" should be the same as "initial=0". I'm not sure if you intended to verify this in this test, if not, you don't need to test it here.

As I understand, "initial=None" was @bhalevy 's concern [here](https://github.com/scylladb/scylladb/pull/23056#discussion_r1973181941)

Reply:
Maybe we need a different test to verify what a missing "initial" means, but I'm not sure if it belongs in the DESC TABLE test. The fact that missing "initial" and "initial=0" are the same thing is important even if you never run DESC TABLE.

Reply:
Also, I still don't understand some of these combinations. What's the point of testing tablets=false but initial=7? Do we even expect the table creation to work without an error why?
If the point of this test is to test DESC TABLE we don't need to check dubious corner cases for CREATE TABLE - if you want to check those, it's a different test.

Reply:
I agree.
I think that currently the query succeeds.
Maybe it shouldn't since it seems to be self-contradictory.  Specfying only `initial` enables tablets implicitly, so it contradicts with `enabled: false`.

Reply:
Note that whether or not Create Table errors for these strange (like tablets disabled but initial 7), it still doesn't mean that a test for Describe table needs to "enshrine" the fact that it works. If one day we decide to outlaw the combination enabled=false initial=7, do we want the describe table test to start failing? I doubt it.

Reply:
Agreed

Reply:
Skipping any tests with `initial` when the `tablets` disabled